src/registry/docsetregistry.cpp

```diff
@@ -143,6 +145,24 @@ void DocsetRegistry::_runQuery(const QString &query)
     emit queryCompleted(results);
 }
 
+SearchQuery DocsetRegistry::getSearchQuery(const QString &queryStr) const
+{
+    return SearchQuery::fromString(queryStr, completions());
+}
+
+QStringList DocsetRegistry::completions() const
```

Comment:
The completion list can be built in runtime as docsets are added and removed. Should be more efficient.